package/etc/pylib/config_generator/addons.py

```diff
@@ -0,0 +1,51 @@
+import os
+from logging import getLogger
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List
+
+import yaml
+
+
+logger = getLogger(__name__)
+
+
+@dataclass
+class AddonMetada:
+    name: str
+
+
+@dataclass
+class Addon:
+    path: Path
+    metadata: AddonMetada
+
+
+def load_addons(addons_directory: Path) -> List[Addon]:
```

Comment:
you may but don't have to consider an alternative draft:

```

def load_addons(addons_directory: Path) -> List:
    addon_candidates = [
        candidate for candidate in addons_directory.iterdir()
        if candidate.is_dir() and (candidate / "addon_metadata.yaml").exists()
    ]

    addons = []
    for candidate in addon_candidates:
        try:
            addon_metadata = load_addon_metadata(candidate)
            addons.append(addon_metadata)
        except Exception as e:
            logger.error(f"Failed to load addon {candidate}: {e}")
            raise

    return addons
```

Reply:
done